### ISSUE TITLE
[2.1.5] Ubuntu16.04 directly invoking waagent command is broken

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/WALinuxAgent#322]
Ubuntu16.04

```
# waagent -version
Traceback (most recent call last):
  File "/usr/sbin/waagent", line 31, in <module>
    import azurelinuxagent.agent as agent
ImportError: No module named azurelinuxagent.agent
```

This should be working as people invoke waagent (/usr/sbin/waagent) directly (e.g. `waagent -deprovision+user`)

I'm guessing this is because we install the agent to Ubuntu 16.04 with `python3 setup.py install` and /usr/sbin/waagent has `#!/usr/bin/env python` (not python3)...

Not sure how to solve this? Not a blocking issue most probably, as `python3 /usr/sbin/waagent -version` still works...

**Comment by brendandixon** [comment]
@ahmetalpbalkan None of that code has been changed by our recent work. Further, I understand that the distro vendors own how the agent gets installed?

**Comment by ahmetalpbalkan** [comment]
should be fixed nonetheless. some of our docs on the Azure Documentation website instructs people to run `waagent -...` commands.

**Comment by brendandixon** [comment]
@ahmetalpbalkan I believe your 16.04 is not in a healthy state. On a clean 16.04 image, I can run /usr/sbin/waagent without issue.

```bash
root@ubuntu1604:~# /usr/sbin/waagent -daemon
2016/07/18 20:12:27.179199 INFO Azure Linux Agent Version:2.1.3
2016/07/18 20:12:27.185913 INFO OS: ubuntu 16.04
2016/07/18 20:12:27.191166 INFO Python: 3.5.1
2016/07/18 20:12:27.196139 INFO Daemon is already running: 1254
```

**Comment by ahmetalpbalkan** [comment]
@brendandixon issues is not with clean 16.04. It is about 2.1.5 as the title suggests.

Please run the following cmds on your clean VM:

```
# waagent -version
WALinuxAgent-2.1.3 running on ubuntu 16.04
Python: 3.5.1
```

```
wget https://bootstrap.pypa.io/ez_setup.py -O - | python3

git clone https://github.com/Azure/WALinuxAgent

cd WALinuxAgent

python3 setup.py install
```

```
# waagent -version
Traceback (most recent call last):
  File "/usr/sbin/waagent", line 31, in <module>
    import azurelinuxagent.agent as agent
ImportError: No module named azurelinuxagent.agent

# /usr/sbin/waagent -daemon
Traceback (most recent call last):
  File "/usr/sbin/waagent", line 31, in <module>
    import azurelinuxagent.agent as agent
ImportError: No module named azurelinuxagent.agent
```

**Comment by brendandixon** [comment]
@ahmetalpbalkan The problem is that, after install, waagent is launched with Python2 vs. Python3. I suspect the setup.py code (or something related) was never properly written for Python3.

So, yes, we should address this, but, again, on a clean image with the 2.1.x branch installed correctly by the distribution vendor, this problem will not occur.

**Comment by brendandixon** [comment]
@ahmetalpbalkan This problem occurs with the 2.1.3 branch if installed manually.

**Comment by brendandixon** [comment]
@ahmetalpbalkan This is, as I originally suggested, not a bug. The default version of bin/waagent begins with:

```bash
#!/usr/bin/env python
```

This is correct for *most* distributions. However, 16.04 *prefers* Python3. The vendor modifies `/usr/sbin/waagent` so it begins with:

```bash
#!/usr/bin/env python3
```

As I suggested at the start: This is an issue arising from manually installing the agent. If a user wants to install the agent under Python3, then they must also update the `#!` portion of `/usr/sbin/waagent` or invoke it directly with Python3.

**Comment by ahmetalpbalkan** [comment]
@brendandixon got it. So leaving this here perhaps we can find a way to fix it downstream...

**Comment by brendandixon** [comment]
@ahmetalpbalkan Doubtful. It's a Python problem. The packages are installed per version. You must launch dependent code with that version. We could, perhaps, document it, but no more.

Welcome to Python3-land.

**Comment by ahmetalpbalkan** [comment]
Understood, looks like something we can't entirely fix on 16.04.
